### PR TITLE
[WIP] Add modified & simple test

### PR DESCRIPTION
**Pull request by limx0**
- Add `modified` method in conjunction with  https://github.com/intake/filesystem_spec/pull/268

### CODE REVIEW

File: s3fs/core.py [hunk 1]
```diff
@@ -1040,6 +1040,10 @@ def walk(self, path, maxdepth=None, **kwargs):
             raise ValueError('Cannot crawl all of S3')
         return super().walk(path, maxdepth=maxdepth, **kwargs)
 
+    def modified(self, path, version_id=None, refresh=False):
+        info = self.info(path=path, version_id=version_id, refresh=refresh)
+        return info['LastModified'].replace(tzinfo=None)
```
Comment (martindurant) [inline s3fs/core.py:1045]: Buckets have `'CreationDate'` and deeper directories have neither field. I don't know if you intend this to work on directories, but they `KeyError` may come as a surprise.

Comment (limx0) [reply s3fs/core.py:1045]: My intention was only to support actual files, not directories (or s3s pseudo directories). Let me push an update.

File: s3fs/core.py [hunk 2]
```diff
@@ -1042,6 +1042,9 @@ def walk(self, path, maxdepth=None, **kwargs):
 
     def modified(self, path, version_id=None, refresh=False):
         info = self.info(path=path, version_id=version_id, refresh=refresh)
+        if 'LastModified' not in info:
+            # This path is a bucket or folder, which do not currently have a modified date
+            return
```
Comment (martindurant) [inline s3fs/core.py:1048]: Seems a little odd to just ignore. Could raise `IsADirectoryError`?

Comment (limx0) [reply s3fs/core.py:1048]: Yep - no strong opinion here @martindurant, let me add one.

**Comment by martindurant** [comment]
Thanks, this is going in.

**Pull request merged**